bundle/config/variable/variable.go

```diff
@@ -66,19 +65,25 @@ func (v *Variable) Set(val VariableValue) error {
 		return fmt.Errorf("variable has already been assigned value: %s", v.Value)
 	}
 
-	rv := reflect.ValueOf(val)
-	switch rv.Kind() {
-	case reflect.Struct, reflect.Array, reflect.Slice, reflect.Map:
-		if v.Type != VariableTypeComplex {
-			return errors.New("variable type is not complex")
-		}
+	if v.IsComplexValued() && v.Type != VariableTypeComplex {
+		return fmt.Errorf("variable type is not complex: %s", v.Type)
 	}
 
 	v.Value = val
 
 	return nil
 }
 
+func (v *Variable) IsComplexValued() bool {
```

Comment:
What's the reason this check is not done internally in IsComplex()?

Reply:
That's used in a separate place to validate that the type annotation is set for variable overrides (in some cases like from a variable-overrides.json file). The goal of this PR is to track telemetry, not to try and remove this annotation check.